genie-agent/src/test/groovy/com/netflix/genie/agent/execution/services/impl/FileSystemFetchingCacheServiceSpec.groovy

```diff
@@ -0,0 +1,287 @@
+                temporaryFolder.newFile("bin.tar.gz")
+        ]
+        Resource[] resources = [
+                Mock(Resource),
+                Mock(Resource),
+                Mock(Resource)
+        ]
+
+        when:
+        cache.get(Sets.newHashSet([
+                Pair.of(uris[0], targetFiles[0]),
+                Pair.of(uris[1], targetFiles[1]),
+                Pair.of(uris[2], targetFiles[2]),
+        ]))
+
+        then:
+        1 * resourceLoader.getResource(uris[0].toString()) >> resources[0]
+        1 * resourceLoader.getResource(uris[1].toString()) >> resources[1]
+        1 * resourceLoader.getResource(uris[2].toString()) >> resources[2]
+        1 * resources[0].exists() >> true
+        1 * resources[1].exists() >> true
+        1 * resources[2].exists() >> true
+        1 * resources[0].lastModified() >> DEFAULT_RESOURCE_LAST_MODIFIED_TS
+        1 * resources[1].lastModified() >> DEFAULT_RESOURCE_LAST_MODIFIED_TS
+        1 * resources[2].lastModified() >> DEFAULT_RESOURCE_LAST_MODIFIED_TS
+        1 * resources[0].getInputStream() >> new ByteArrayInputStream(uris[0].toString().getBytes(StandardCharsets.UTF_8))
+        1 * resources[1].getInputStream() >> new ByteArrayInputStream(uris[1].toString().getBytes(StandardCharsets.UTF_8))
+        1 * resources[2].getInputStream() >> new ByteArrayInputStream(uris[2].toString().getBytes(StandardCharsets.UTF_8))
+        targetFiles[0].getText(StandardCharsets.UTF_8.toString()) == uris[0].toString()
+        targetFiles[1].getText(StandardCharsets.UTF_8.toString()) == uris[1].toString()
+        targetFiles[2].getText(StandardCharsets.UTF_8.toString()) == uris[2].toString()
+    }
+
+    def "Construct: fail to create cache dir"() {
+        setup:
+        ArgumentDelegates.CacheArguments badCacheArguments = Mock()
+        // This test assumes this directory does not exist and it will fail to be created
+        File cacheDir = new File("/", "genie")
+
+        when:
+        new FileSystemFetchingCacheService(resourceLoader, badCacheArguments, lockingFactory)
+
+        then:
+        1 * badCacheArguments.getCacheDirectory() >> cacheDir
+        thrown(RuntimeException)
+    }
+
+    def "Construct: cache dir does not exist"() {
+        setup:
+        ArgumentDelegates.CacheArguments badCacheArguments = Mock()
+
+        when:
+        new FileSystemFetchingCacheService(resourceLoader, badCacheArguments, lockingFactory)
+
+        then:
+        1 * badCacheArguments.getCacheDirectory() >> new File(temporaryFolder.getRoot(), "genie-cache")
+    }
+
+    def "Construct: cache dir is a file"() {
+        setup:
+        ArgumentDelegates.CacheArguments badCacheArguments = Mock()
+
+        when:
+        new FileSystemFetchingCacheService(resourceLoader, badCacheArguments, lockingFactory)
+
+        then:
+        1 * badCacheArguments.getCacheDirectory() >> temporaryFolder.newFile()
+        thrown(RuntimeException)
+
+    }
+}
+
```

Comment:
Is there a reason there are about 20 empty lines at the end of this file?

Reply:
fixed